shared/client/util.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/chanzuckerberg/happy-shared/model"
+	"github.com/pkg/errors"
+)
+
+var ErrRecordNotFound = errors.New("record not found")
+
+func ParseResponse[T interface{}](resp *http.Response, result *T) error {
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "failed to read response body")
+	}
+
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		return errors.Wrap(err, "failed to unmarshal response body")
+	}
```

Comment:
```suggestion
	err = json.NewDecoder(resp.Body).decode(&result)
	if err != nil {
		return errors.Wrap(err, "failed to unmarshal response body")
	}
```

Making a decoding is more efficent. It allows you to decode the JSON as you are reading the HTTP stream rather than reading all the bytes from the stream and then reading them all again to do the JSON decoding.